lib/TerminalDisplay.cpp

```diff
@@ -2123,8 +2145,45 @@ QList<QAction*> TerminalDisplay::filterActions(const QPoint& position)
   return spot ? spot->actions() : QList<QAction*>();
 }
 
+void TerminalDisplay::unclutter() const
+{
+  if (gs_deadSpot.x() < 0 && underMouse())
+  {
+    gs_deadSpot = gs_futureDeadSpot;
+    QApplication::setOverrideCursor(Qt::BlankCursor);
+  }
+}
+
+void TerminalDisplay::autoHideMouseAfter(int delay)
+{
+  if (delay > -1 && !gs_hideMouseTimer)
+  {
+    gs_hideMouseTimer = new QTimer;
+    gs_hideMouseTimer->setSingleShot(true);
+  }
+  if ((_mouseAutohideDelay < 0) == (delay < 0))
+  {
+    _mouseAutohideDelay = delay;
+    return;
+  }
+  if (delay < 0)
+    disconnect(gs_hideMouseTimer, &QTimer::timeout, this, &TerminalDisplay::unclutter);
+  else
+    connect(gs_hideMouseTimer, &QTimer::timeout, this, &TerminalDisplay::unclutter);
+  _mouseAutohideDelay = delay;
+}
+
 void TerminalDisplay::mouseMoveEvent(QMouseEvent* ev)
 {
+  // unclutter
+  if (gs_deadSpot.x() > -1 && (ev->pos() - gs_deadSpot).manhattanLength() > 8)
+  {
+    gs_deadSpot = QPoint(-1,-1);
+    QApplication::restoreOverrideCursor();
+  }
+  gs_futureDeadSpot = ev->pos();
+  gs_hideMouseTimer->start(_mouseAutohideDelay);
```

Comment:
Missing nullity check.

Reply:
The entire block should be guarded by _mouseAutohideDelay > -1 and I'll add a "Q_ASSERT(gs_hideMouseTimer);"